src/TraceEvent/Computers/ActivityComputer.cs

```diff
@@ -1145,7 +1145,7 @@ private void ResolveWellKnownSymbols()
                 }
             }
             if (!foundThreadingAPIs)
-                throw new ApplicationException("Could not resolve symbols for Task library (mscorlib), task stacks will not work.");
+                throw new Exception("Could not resolve symbols for Task library (mscorlib), task stacks will not work.");
```

Comment:
In PerfView I use ApplicationException for things that you want a user to see.   Things that are not ApplicationExceptions are assumed to be programmer errors and are not surfaced to the end user, but simply say that PerfView had an internal error.    By changing this you lose this distinction.  

I am assuming that ApplicationException is not in V1.6?   

I am OK having an #ifdef that makes it throw Exception if you are on V1.6 but ApplicationException on desktop.

Reply:
That's correct. ApplicationException is not available in 1.6. What if we created a new exception type and threw that? 

That way the developer using TraceEvent can catch that Exception and know it was one of these errors. I notice we catch ApplicationException in a few places within PerfView

Reply:
Make up a new type called ApplicationException in the Microsoft.Diagnostics.Tracing.  Have this exception inherit from the System.Application when we are on desktop (plumb through a new define, let me know if you need help doing this).     

This way on Desktop, code will simply just work.

Reply:
Should we maybe call it "TraceEventApplicationException" or something like that to avoid someone ripping out their hair in the future trying to catch our ApplicationException from a netstandard compiled package when they are on desktop?

Reply:
My goal is to avoid making the code 'worse' for the sake of old things (V1.6 support).   I don't mind if old things have poorer functionality.    My personal opinion is that you should not have many different exception types (who uses them?) but you doe need a few 'groups' and this idea that knowing just the exception type you can usefully predict if it is likely some human should see.  

I really don't want to make users do 'wierd' things for V2.0 or beyond (I don't care about 1.6), and frankly ANY changes are suspect (as this one is).  

One 'easy' way of doing this to have a V2.0 and a V1.6 builds in the nuget package   Then simply have ifdefs for the V1.6 version.   This will also be useful for the issue you ran into with elevation, and I suspect we might run into it in the future as well.   

What do you think about having a helper that throws the applicationException and having #ifdef code that throws Exception on V1.6 in the body?